### PR TITLE
Add Company object

### LINKED ISSUE #41: Add Company Object
- Add Company Object to Connectify
- Company Object should contain various attributes, namely company's name, industry, location, description, website, email, phone and address
- Add test cases for Company Object

### PR DESCRIPTION
**Pull request by geraldngjx**
This PR fixes #41 .

Notable changes:
- Refactor seedu.address naming to connectify
- Add Company Object to Connectify
- Company Object contains various attributes, namely company's name, industry, location, description, website, email, phone and address
- Add test cases for Company Object
- Fix existing Checkstyle issues

### CODE REVIEW

File: src/main/java/connectify/model/company/Company.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
+package connectify.model.company;
+
+import static connectify.commons.util.CollectionUtil.requireAllNonNull;
+
+/**
+ * Represents a Company in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Company {
```
Comment (nreHieW) [inline src/main/java/connectify/model/company/Company.java:11]: Clean use of code 👍

File: src/main/java/connectify/model/company/Company.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+     * @param location Location of company
+     * @param description Description of company
+     * @param website Website of company
+     * @param email Email of company
+     * @param phone Phone of company
+     * @param address Address of company
+     */
+    public Company(String name, String industry, String location, String description, String website, String email,
+                   String phone, String address) {
+        requireAllNonNull(name, industry, location, description, website, email, phone, address);
```
Comment (nreHieW) [inline src/main/java/connectify/model/company/Company.java:34]: Good check of input validation

File: src/test/java/connectify/model/company/CompanyTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import static connectify.testutil.TypicalCompanies.COMPANY_1;
+import static connectify.testutil.TypicalCompanies.COMPANY_2;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import connectify.testutil.CompanyBuilder;
+
+public class CompanyTest {
```
Comment (nreHieW) [inline src/test/java/connectify/model/company/CompanyTest.java:13]: Very comprehensive use of test cases to check all dependencies

**Review by nreHieW — APPROVED** [review]
Good changes however, do add more test cases to ensure higher code coverage

**Review by nreHieW — CHANGES_REQUESTED** [review]
Do add more test cases to ensure better code coverage. 

Specifically, do add more test cases for the hashcode and the string representation of the Company class.

File: src/main/java/connectify/model/company/Company.java (hunk 1 above)
Comment (geraldngjx) [reply src/main/java/connectify/model/company/Company.java:11]: Thank you!

**Comment by geraldngjx** [comment]
Notable Changes:
Add test cases to check for the functionality of the Company hashCode function and toString function.

**Review by nreHieW — APPROVED** [review]
Thanks for adding more test cases as per my request! LGTM 👍

**Pull request merged**